src/Gaffer/MetadataAlgo.cpp

```diff
@@ -222,6 +224,89 @@ void bookmarks( const Node *node, std::vector<NodePtr> &bookmarks )
 	}
 }
 
+void setNumericBookmark( Node *node, int value )
+{
+	if( value < 0 )
+	{
+		return;
```

Comment:
If the value is out of the valid range, an exception would be more appropriate than silent failure.